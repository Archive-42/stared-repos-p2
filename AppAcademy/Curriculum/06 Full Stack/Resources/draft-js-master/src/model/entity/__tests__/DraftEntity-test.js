/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @emails oncall+draft_js
 * @flow strict-local
 * @format
 */

'use strict';

jest.disableAutomock();

const DraftEntity = require('DraftEntity');

beforeEach(() => {
  jest.resetModules();
});

const createLink = () => {
  return DraftEntity.__create('LINK', 'MUTABLE', {uri: 'zombo.com'});
};

test('must create instances', () => {
  const key = createLink();
  expect(typeof key).toMatchSnapshot();
});

test('must retrieve an instance given a key', () => {
  const key = createLink();
  const retrieved = DraftEntity.__get(key);
  expect(retrieved.getType()).toMatchSnapshot();
  expect(retrieved.getMutability()).toMatchSnapshot();
  expect(retrieved.getData()).toMatchSnapshot();
});

test('must throw when retrieving for an invalid key', () => {
  createLink();
  expect(() => DraftEntity.__get('asdfzxcvqweriuop')).toThrow();
});

test('must merge data', () => {
  const key = createLink();

  // Merge new property.
  const newData = {foo: 'bar'};
  DraftEntity.__mergeData(key, newData);
  const newEntity = DraftEntity.__get(key);

  // Replace existing property.
  const withNewURI = {uri: 'homestarrunner.com'};
  DraftEntity.__mergeData(key, withNewURI);
  const entityWithNewURI = DraftEntity.__get(key);

  expect(newEntity.getData()).toMatchSnapshot();
  expect(entityWithNewURI.getData()).toMatchSnapshot();
});
