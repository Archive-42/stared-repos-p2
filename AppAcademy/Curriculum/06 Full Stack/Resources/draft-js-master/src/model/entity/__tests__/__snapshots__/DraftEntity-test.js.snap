// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must create instances 1`] = `"string"`;

exports[`must merge data 1`] = `
Object {
  "foo": "bar",
  "uri": "zombo.com",
}
`;

exports[`must merge data 2`] = `
Object {
  "foo": "bar",
  "uri": "homestarrunner.com",
}
`;

exports[`must retrieve an instance given a key 1`] = `"LINK"`;

exports[`must retrieve an instance given a key 2`] = `"MUTABLE"`;

exports[`must retrieve an instance given a key 3`] = `
Object {
  "uri": "zombo.com",
}
`;
